modules/viz/src/vizimpl.cpp

```diff
@@ -188,6 +190,33 @@ void cv::viz::Viz3d::VizImpl::spinOnce(int time, bool force_redraw)
 }
 
 /////////////////////////////////////////////////////////////////////////////////////////////
+void cv::viz::Viz3d::VizImpl::setOffScreenRendering()
+{
+    window_->SetOffScreenRendering(1);
+    offScreenMode_ = true;
+}
+
+/////////////////////////////////////////////////////////////////////////////////////////////
+void cv::viz::Viz3d::VizImpl::removeAllLights()
+{
+    renderer_->RemoveAllLights();
+}
+
+/////////////////////////////////////////////////////////////////////////////////////////////
+void cv::viz::Viz3d::VizImpl::addLight(Vec3d position, Vec3d focalPoint, Color color, Color diffuseColor, Color ambientColor, Color specularColor)
+{
+    vtkSmartPointer<vtkLight> light = vtkSmartPointer<vtkLight>::New();
+    light->SetPosition(position.val);
+    light->SetFocalPoint(focalPoint.val);
+    light->SetColor(vtkcolor(color).val);
+    light->SetDiffuseColor(vtkcolor(diffuseColor).val);
+    light->SetAmbientColor(vtkcolor(ambientColor).val);
+    light->SetSpecularColor(vtkcolor(specularColor).val);
```

Comment:
Here is an error, actually reference on an automatic object. You need to store on stack Color instance returned from vtk color.